server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -148,31 +163,82 @@ private Setting<?> createSetting(
                     : Setting.boolSetting(key, (Setting<Boolean>) fallback.getSetting(), propertyArray);
             case Integer:
                 return fallback == null
-                    ? Setting.intSetting(key, (int) defaultValue, propertyArray)
+                    ? (parser instanceof Writeable)
```

Comment:
The ternary operator logic could be a little confusing for a user to follow.  Since there are three outcomes to the conditional logic here, I think an if/else if/else block would be better.

Reply:
sure, I will change to if/else in next commit.